mongoquery/__init__.py

```diff
@@ -54,31 +54,29 @@ def _extract(self, entry, path):
         else:
             return _Undefined()
 
-    def _get_value_at_path(self, obj, path):
-        is_dotted = True if path.find(".") != -1 else False
-        is_indexed = True if path.find("[") != -1 else False
-        if is_indexed:
-            path = path.replace("[", ".")
-            is_dotted = True
-        keys = " ".join(path.split(".")).split() if is_dotted else [ path ]
-        for k in keys:
-            if k.endswith(']'):
-                k = int(k[:-1])
-            if type(obj) is str:
-                raise KeyError("Invalid key '{}' in JSON path".format(k))
-            obj = obj[k]
-        return obj
-
     def _path_exists(self, operator, condition, entry):
-        try:
-            self._get_value_at_path(entry, operator)
-        except:
-            return not condition
+        if type(operator) is str:
+            keys_lists = list(operator.split('.'))
+        else:
+            keys_lists = [ path ]
+        for i, k in enumerate(keys_lists):
+            if isinstance(entry, Sequence) and not k.isdigit():
+                for e in entry:
+                    operator = '.'.join(keys_lists[i:])
+                    if self._path_exists(operator, condition, e) == condition:
+                        return condition
+                return not condition
+            elif isinstance(entry, Sequence):
+                k = int(k)
+            try:
+                entry = entry[k]
+            except:
+                return not condition
         return condition
 
     def _process_condition(self, operator, condition, entry):
         if isinstance(condition, Mapping) and "$exists" in condition:
-            if type(operator) is str and (operator.find('.') != -1 or operator.find('[') != -1):
+            if type(operator) is str and operator.find('.') != -1:
```

Comment:
If this keeps coming up then it might be worth having an is_str function to cover it.